File: js/cm-api-v2/PostWebhook.js
// DocSection: cm_api_v2_post_webhook
// Tip: Find more about JS/TS SDKs at https://docs.kontent.ai/javascript
// Using ES6 syntax
import { ManagementClient } from '@kentico/kontent-management';

const client = new ManagementClient({
  projectId: '<YOUR_PROJECT_ID>',
  apiKey: '<YOUR_API_KEY>'
});

client.addWebhook()
  .withData(
    {
      name: "Example webhook",
      url: "https://example.com/webhook",
      secret: "secret_key",
      triggers: {
        delivery_api_content_changes: [
          {
            type: "content_item_variant",
            operations: [
              "publish",
              "unpublish"
            ]
          },
          {
            type: "taxonomy",
            operations: [
              "archive",
              "restore",
              "upsert"
            ]
          }
        ],
        preview_delivery_api_content_changes: [
          {
            type: "content_item_variant",
            operations: [
              "archive",
              "upsert"
            ]
          },
          {
            type: "taxonomy",
            operations: [
              "archive",
              "restore",
              "upsert"
            ]
          }
        ],
        workflow_step_changes: [
          {
            type: "content_item_variant",
            transitions_to: [
              {
                id: "b4363ccd-8f21-45fd-a840-5843d7b7f008"
              },
              {
                id: "88ac5e6e-1c5c-4638-96e1-0d61221ad5bf"
              }
            ]
          }
        ],
        management_api_content_changes: [
          {
            type: "content_item_variant",
            operations: [
              "archive",
              "create",
              "restore"
            ]
          }
        ]
      }
    }
  )
  .toObservable()
  .subscribe((response) => {
    console.log(response);
  },
  (error) => {
    console.log(error);
  });
// EndDocSection

File: net/management-api-v2/PostWebhook.cs
// DocSection: cm_api_v2_post_webhook
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    ProjectId = "<YOUR_ENVIRONMENT_ID>"
});

var response = await client.CreateWebhookAsync(new WebhookCreateModel
{
    Name = "Example webhook",
    Url = "https://example.com/webhook",
    Secret = "secret_key",
    Triggers = new WebhookTriggersModel
    {
        DeliveryApiContentChanges = new[]
        {
            new DeliveryApiTriggerModel
            {
                Type = TriggerChangeType.LanguageVariant,
                Operations = new []
                {
                    "publish",
                    "unpublish"
                }
            },
            new DeliveryApiTriggerModel
            {
                Type = TriggerChangeType.Taxonomy,
                Operations = new []
                {
                    "archive",
                    "restore",
                    "upsert"
                }
            }
        },
        PreviewDeliveryApiContentChanges = new[]
        {
            new DeliveryApiTriggerModel
            {
                Type = TriggerChangeType.LanguageVariant,
                Operations = new []
                {
                    "archive",
                    "upsert"
                }
            },
            new DeliveryApiTriggerModel
            {
                Type = TriggerChangeType.Taxonomy,
                Operations = new []
                {
                    "archive",
                    "restore",
                    "upsert"
                }
            }
        },
        WorkflowStepChanges = new[]
        {
            new WorkflowStepTriggerModel
            {
                TransitionsTo = new []
                {
                    Reference.ById(Guid.Parse("b4363ccd-8f21-45fd-a840-5843d7b7f008")),
                    Reference.ById(Guid.Parse("88ac5e6e-1c5c-4638-96e1-0d61221ad5bf")),
                }
            },
        },
        ManagementApiContentChanges = new[]
        {
            new ManagementApiTriggerModel
            {
                Operations = new []
                {
                    "archive",
                    "create",
                    "restore",
                }
            }
        },
    }
});

// EndDocSection